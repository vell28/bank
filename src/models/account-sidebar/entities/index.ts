export interface IAccountSidebar {
  [key: string]: boolean;
}
