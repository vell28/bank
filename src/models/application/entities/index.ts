export interface IApplication {
  applicationId: string;
  productId: string;
  Authorization: string;
  deviceId?: string;

  readonly bankName: string;
  readonly BIC: string;
}
