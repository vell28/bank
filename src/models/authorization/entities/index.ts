export interface IAuthorization {
  service_code?: string;
  phone?: string;
  token?: string;
  sessid?: string;
  token_expires?: Date;
  // maybe its not required
  'dev-id'?: string;
  'token-firebase'?: string;
  isShownLogoutConfirmModal: boolean;
}

export interface ITokenRefresh {
  success: boolean;
  token: string;
}

export interface ISuccessLogout {
  success: boolean;
}

export interface ICodeRequest {
  sessid: string;
  code: string | null;
  success?: boolean;
}

export interface IPasswordCheckRequest {
  password: string;
  phone: string;
}

export interface ISuccessCheckResponse {
  status: 'ok';
}
