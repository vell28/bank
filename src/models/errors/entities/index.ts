export type AppErrorType = '401' | '404' | '403' | '500' | 'ERR_INTERNET_DISCONNECTED';

export interface IAppErrors {
  title?: string;
  subtitle?: string;
  onOk?: () => void;
  btnText?: string;
  supportUrl?: string;
  type?: AppErrorType;
  meta?: string;
}
