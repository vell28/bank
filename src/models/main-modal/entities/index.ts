import { AllMainModalContentsType } from 'containers/operations/types';

export type MainModalContentType = AllMainModalContentsType | '';

export interface IMainModalSettings {
  id: MainModalContentType;
  bindUrl: string;
  title?: string;
}

export interface IMainModal {
  isShown: boolean;
  settings: IMainModalSettings;
  step: number;
}
