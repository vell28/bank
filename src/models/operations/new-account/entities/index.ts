export enum AccountTypes {
  LINKED = 'LINKED',
  BUSINESS = 'BUSINESS',
  SAVINGS = 'SAVINGS',
}

export type AccountTypeType =
  | AccountTypes.LINKED
  | AccountTypes.BUSINESS
  | AccountTypes.SAVINGS;

export enum AccountNames {
  LINKED = 'Linked',
  BUSINESS = 'Business',
  SAVINGS = 'Savings',
}

export type AccountNameType =
  | AccountNames.LINKED
  | AccountNames.BUSINESS
  | AccountNames.SAVINGS;

export interface IAccount {
  type: AccountTypeType;
  name: AccountNameType;
}

export const ACCOUNTS: IAccount[] = [
  {
    type: AccountTypes.LINKED,
    name: AccountNames.LINKED,
  },
  {
    type: AccountTypes.BUSINESS,
    name: AccountNames.BUSINESS,
  },
  {
    type: AccountTypes.SAVINGS,
    name: AccountNames.SAVINGS,
  },
];
