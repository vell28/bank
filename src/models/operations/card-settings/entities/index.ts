import { CurrencyCodeType } from 'modules/currencies';
import { AccountStatusType, IBalance } from 'models/organizations/entities';

export enum CardProduct {
  MAIN = 'MAIN',
  ADDITIONAL = 'ADDITIONAL',
  FAMILY = 'FAMILY',
  KIDS = 'KIDS',
}

export type CardProductType = CardProduct.MAIN | CardProduct.ADDITIONAL | CardProduct.FAMILY | CardProduct.KIDS;

export interface ICardOwner {
  embossedName: string;
}

export enum Limit {
  DAY = 'DAY',
  MONTH = 'MONTH',
  FREE_ATM_WITHDRAWAL = 'FREE_ATM_WITHDRAWAL',
}

export type LimitType = Limit.DAY | Limit.MONTH | Limit.FREE_ATM_WITHDRAWAL;

export interface ILimit {
  type: LimitType;
  title: null | string;
  maxValue: number;
  usedValue: number;
  preferredMaxValues?: number[];
}

export interface ILimitData {
  type: LimitType;
  maxValue: string;
}

export interface ICardOptions {
  limits?: {
    disable?: boolean;
    disable5min?: boolean;
    disableOneTime?: boolean;
  };
}

export interface ICard {
  accountId: string;
  availableBalance: number;
  backgroundImage?: string;
  balance: number;
  balanceHold: number;
  balances: IBalance[];
  clientId: string;
  createdAt?: string;
  currency: {
    code: CurrencyCodeType;
  };
  expireAt?: string;
  id: string;
  limits: ILimit[];
  number?: string;
  options: ICardOptions;
  owner: ICardOwner;
  paySystem: string;
  plasticStatus: string;
  productType: CardProductType;
  status: AccountStatusType;
  title?: string;
}

export interface ICardUnmasked {
  number: string;
  pin: string | null;
  cvc: string;
  expireAt: string;
  owner: ICardOwner;
}
