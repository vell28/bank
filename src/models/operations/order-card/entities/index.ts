import { CardProductType, ICardOwner } from '../../card-settings/entities/index';

export interface ICardOrderOwner extends ICardOwner {
  phoneNumber: string;
}

export interface IAddress {
  apartmentNumber: number;
  city: string;
  countryCode: string;
  postalCode: string;
  province?: string;
  street: string;
  streetNumber: string;
}

export interface ICardOrder {
  cardProductType?: CardProductType;
  cardOwner?: ICardOrderOwner;
  deliveryAddress?: IAddress;
}

export enum CardFormat {
  VIRTUAL = 'VIRTUAL',
  PLASTIC = 'PLASTIC',
}

export type CardFormatType = CardFormat.VIRTUAL | CardFormat.PLASTIC;

export interface IOrderRequestBody {
  type: CardProductType;
  format: CardFormatType;
  cardHolderName: string;
  cardHolderPhoneNumber: string;
  isExpressDelivery: boolean;
  deliveryAddress: IAddress;
}
