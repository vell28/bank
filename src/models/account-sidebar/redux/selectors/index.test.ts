import { emptyStore } from 'modules/store/emptyStore';
import { getAccountSidebarMenu, getIsShownSidebar } from '.';

import { IAccountSidebar } from '../../entities';

const accountId = 'accountId';

const data: IAccountSidebar = {
  [accountId]: true,
};

const store = {
  ...emptyStore,
  accountSidebar: {
    data,
    isShown: true,
  },
};

describe('account  sidebar selector tests', () => {
  test('getAccountSidebarMenu should return menu', () => {
    const sidebar = getAccountSidebarMenu(store);
    expect(sidebar[accountId]).toBeTruthy();
    expect(sidebar).toMatchSnapshot();
  });

  test('getIsShownSidebar should return isShown menu', () => {
    const isShown = getIsShownSidebar(store);
    expect(isShown).toBeTruthy();
  });
});
