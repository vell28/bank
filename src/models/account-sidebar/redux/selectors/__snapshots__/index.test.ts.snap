// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account  sidebar selector tests getAccountSidebarMenu should return menu 1`] = `
Object {
  "accountId": true,
}
`;
